HIP-0013.md

```diff
@@ -0,0 +1,130 @@
+
+## TXT Specification
+
+IDNS TXT records begin with **`IDNS1`** (version 1), and use the following format containing a record type, key, and value: 
+
+<pre><code><b>IDNS1 &lt;type&gt;:&lt;key&gt;=&lt;value&gt;</b>
+</code></pre>
+
+IDNS TXT records can contain one or more records in key=value pairs, separated by spaces:
+<pre><code><b>IDNS1</b> profile:avatar=<ins>http://pic.com/me.png</ins> service:com.twitter=<ins>aoxborrow</ins>
+</code></pre>
+
+Records that contain spaces must be single quoted. Single quotes in the record can be escaped with a backslash:
+<pre><code><b>IDNS1</b> profile:location=<ins>'Berkeley, CA'</ins> profile:bio=<ins>'I\'m a passionate advocate for domain innovation.'</ins></code></pre>
+
+### Profile Records (`profile`)
+Human-readable user profile text data. 
+  - **`<key>`** The profile record name, any text key is valid, no spaces.
+  - **`<value>`** The profile record data.
+<pre><code><b>IDNS1 profile:&lt;key&gt;=&lt;value&gt;</b>
+
+<i>Examples:</i>
+IDNS1 profile:name=aox
+IDNS1 profile:avatar=https://hns.id/domain/aox.hns/avatar.png
+IDNS1 profile:location='Berkeley, CA'
+IDNS1 profile:email=hey@me.com
+</code></pre>
+
+### Service Records (`service`)
+Service keys must be made up of a reverse dot notation for a namespace which the service owns, for example, DNS names (e.g. .com, .io) or HNS name (i.e. .hns). Service Keys must contain at least one dot. See [EIP-634](https://eips.ethereum.org/EIPS/eip-634#service-keys).
+  - **`<key>`** A service key associated with the user, e.g. `com.github`.
+  - **`<identifier>`** Identifier for the service, e.g. Github username. 
+<pre><code><b>IDNS1 service:&lt;key&gt;=&lt;identifier&gt;</b>
+
+<i>Examples:</i>
+IDNS1 service:com.twitter=aoxborrow
+IDNS1 service:com.github=0xstefan
+IDNS1 service:io.keybase=pinheadmz
+</code></pre>
+
+### Wallet Records (`wallet`)
+Cryptocurrency wallet addresses. Uses [SLIP44](https://github.com/satoshilabs/slips/blob/master/slip-0044.md) to specify different blockchains.
+  - **`<cointype>`** The type of cryptocurrency, using SLIP44 cointype symbol or number, e.g. `hns` or `5353`
+  - **`<address>`** Wallet address for the specific cryptocurrency.
+<pre><code><b>IDNS1 wallet:&lt;cointype&gt;=&lt;address&gt;</b>
+
+<i>Examples:</i>
+IDNS1 wallet:btc=1Dvf5aQkVrVJeasRUDiLATGWLBGFWA9ovY
+IDNS1 wallet:hns=hs1qshuyulxra3pqpwr40303t8pn79232zztuk4qgz
+IDNS1 wallet:5353=hs1qshuyulxra3pqpwr40303t8pn79232zztuk4qgz
+IDNS1 wallet:eth=0xa08abA528da538cd3547a73C93d51B90201294B6
+</code></pre>
+
+
+### Contenthash Records (`contenthash`)
+A content hash or address for distributed networks like IPFS. See [ENSIP-7](https://docs.ens.domains/ensip/7).
+  - **`<protocol>`** Distributed network protocol name, e.g. `ipfs`.
```

Comment:
I haven't worked with contenthash much but the idea was just to put something human readable, and basically a placeholder, since otherwise this record type has no key/param

Reply:
I put a note that the `<protocol>` key is not used for resolution. I think it's kind of nice for human eyes but we could remove the key entirely 

@pcfreak30 do you know if ENSIP-7 `mulicodec` is widely adopted? do you know of any reason not to use it?

Reply:
^^ @hodlthedoor same question

Reply:
I just wanted to bring it up since it can save a few bytes on chain, but no strong opinions either way.
If the protocol is removed, it would look like (taking IPFS example from ENSIP-7):
```
IDNS1 contenthash:ipfs=0xe3010170122029f2d17be6139079dc48696d1f582a8530eb9805b561eda517e22a892c7e3f1f
```
And the `0xe3` prefix indicates IPFS. So instead of matching `txt.startsWith("ipfs=")`, it becomes `txt.startsWith("0xe3")`.

But reading more of that proposal, since we are dealing with TXT records here, the multicodec hex is being saved as a string, which only makes it longer and take up more space! Simple text format (again example taken from ENSIP-7) is a lot shorter:
```
IDNS1 contenthash:ipfs://QmRAQB6YaCyidP37UdDnjFY5vQuiBrcqdyoW1CuDgwxkD4
```

hex would make sense if we are writing the bytes directly like a new record type IDNS1, but since TXT is string only, imo the text form is clearer and shorter.

---

Regardless of what's decided above, I think we should use real examples, at least of the same length as what will be expected.